src/coreclr/jit/fgopt.cpp

```diff
@@ -260,20 +260,17 @@ void Compiler::fgComputeReachabilitySets()
         {
             BasicBlock* const block = fgBBReversePostorder[i];
 
-            bool predGcSafe = (block->bbPreds != nullptr); // Do all of our predecessor blocks have a GC safe bit?
+            BasicBlockFlags predGcFlags = BBF_GC_SAFE_POINT; // Do all of our predecessor blocks have a GC safe bit?
 
             for (BasicBlock* const predBlock : block->PredBlocks())
             {
                 /* Union the predecessor's reachability set into newReach */
                 change |= BlockSetOps::UnionDChanged(this, block->bbReach, predBlock->bbReach);
 
-                if (!(predBlock->bbFlags & BBF_GC_SAFE_POINT))
-                {
-                    predGcSafe = false;
-                }
+                predGcFlags &= predBlock->bbFlags;
             }
 
-            if (predGcSafe)
+            if ((block->bbPreds != nullptr) && (predGcFlags != BBF_EMPTY))
```

Comment:
Avoid a few more branches?

```suggestion
            block->bbFlags |= predGcFlags;
```

Reply:
We have to check `block->bbPreds != nullptr` or else we'd set any block with no predecessors (commonly the first block) with `BBF_GC_SAFE_POINT` unconditionally. But true, we don't need to check against `BBF_EMPTY`.